jwst/pipeline/calwebb_guider.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python
+from ..stpipe import Pipeline
+from .. import datamodels
+import os
+
+# step imports
+from ..dq_init import dq_init_step
+from ..flatfield import flat_field_step
+from ..guider_cds import guider_cds_step
+
+__version__ = "7.0.0"
+
+# Define logging
+import logging
+log = logging.getLogger()
+log.setLevel(logging.DEBUG)
+
+class GuiderPipeline(Pipeline):
+    """
+    GuiderPipeline: For FGS observations, apply all calibration 
+    steps to raw JWST ramps to produce a 3-D slope product. 
+    Included steps are: dq_init, flat_field, and guider_cds.
+    """
+
+    spec = """
+        save_calibrated_ramp = boolean(default=False)
+    """
+
+    # Define aliases to steps
+    step_defs = {'dq_init': dq_init_step.DQInitStep,
+                 'flat_field': flat_field_step.FlatFieldStep,
+                 'guider_cds': guider_cds_step.GuiderCdsStep,
+                 }
+
+    # start the actual processing
+    def process(self, input):
+
+        log.info('Starting calwebb_guider ...')
+
+        # open the input
+        input = datamodels.open(input)
```

Comment:
I wonder if we need to explicitly open as an FGS/Guider data model, rather than using the generic .open() and relying on it to figure out what kind of data model to load it into. It might not be very good at recognizing this unique case.